src/main/java/com/google/sps/DataServlet.java

```diff
@@ -153,8 +162,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     if (mutationNumber > currDataGraph.numMutations()) {
       diff = Utility.getMultiMutationAtIndex(mutList, mutationNumber);
     }
+    List<String> nodeNames = new ArrayList<>();
+    try {
+      JsonArray nodeNameArr = JsonParser.parseString(nodeNamesParam).getAsJsonArray();
+      for (int i = 0; i < nodeNameArr.size(); i++) {
+        String curr = nodeNameArr.get(i).getAsString().trim();
+        if (curr.length() > 0) {
+          nodeNames.add(curr);
+        }
+      }
+    } catch (JsonSyntaxException | IllegalStateException e) {
+    }
+
+    // A list of "roots" to return nodes at most depth radius from
+    HashSet<String> queried = new HashSet<>();
+
+    // roots to calculate the mutations from
+    HashSet<String> queriedNext = new HashSet<>();
+    // We start by adding the node name if it was searched for
+    if (nodeNames.size() > 0) {
+      queried.addAll(nodeNames);
+      queriedNext.addAll(nodeNames);
```

Comment:
What if the some of the nodeNames were deleted in this mutation? Should queriedNext still contain these nodes?

Reply:
even if something in nodeNames was not in the graph, would you still want mutations that pertain to it?

Reply:
I guess this was just to make sure that if a node in nodeNames was deleted but then added later on (or just in general added after the current step) the tool would still show it.